lib/osf-components/addon/components/osf-layout/registries-side-nav/icon/component.ts

```diff
@@ -0,0 +1,19 @@
+import { tagName } from '@ember-decorators/component';
+import Component from '@ember/component';
+import { assert } from '@ember/debug';
+
+import { layout } from 'ember-osf-web/decorators/component';
+
+import styles from './styles';
+import template from './template';
+
+@tagName('')
+@layout(template, styles)
+export default class Icon extends Component {
+    icon!: string;
+
+    didReceiveAttrs() {
+        assert('OsfLayout::RegistriesSideNave::Icon: ' +
```

Comment:
Maybe we can do something like this to keep the string all on one line?
```suggestion
        assert(
            'OsfLayout::RegistriesSideNave::Icon: @icon is required for this component to render',
            Boolean(this.icon),
        );
```